File: engineering/codebase/smartqr.frontend-services/src/domain/codes/content/models/UrlContent.ts
import { ContentType } from "../enums/ContentType";

/** Represents the dynamic URL-forwarder content — the QR carries the redirect short link to this destination. */
export interface UrlContent {
  /** The content-type discriminator. */
  type: typeof ContentType.Url;

  /** The destination URL the short link forwards to. */
  url: string;
}

File: engineering/codebase/smartqr.backend-services/SmartQr.Domain/Codes/Content/Url/Models/UrlContent.cs
namespace SmartQr.Domain.Codes.Content.Url.Models;

/// <summary>URL content — a plain destination fronting a dynamic redirect; the symbol bakes no payload.</summary>
public sealed record UrlContent : CodeContent
{
    /// <summary>The destination the redirect forwards to.</summary>
    public required string Url { get; init; }

    /// <inheritdoc />
    public override string? Encode() => null;
}
